compiler/rustc_fs_util/src/lib.rs

```diff
@@ -102,3 +104,45 @@ pub fn path_to_c_string(p: &Path) -> CString {
 pub fn try_canonicalize<P: AsRef<Path>>(path: P) -> io::Result<PathBuf> {
     fs::canonicalize(&path).or_else(|_| absolute(&path))
 }
+
+pub struct TempDirBuilder<'a, 'b> {
+    builder: tempfile::Builder<'a, 'b>,
+}
+
+impl<'a, 'b> TempDirBuilder<'a, 'b> {
+    pub fn new() -> Self {
+        let mut builder = tempfile::Builder::new();
+        builder.rand_bytes(8);
```

Comment:
Why are we deviating from the default `Builder` behavior here?

Reply:
I figured a couple of more random bytes might reduce the chances of whatever is causing the error. But I'm not sure if it actually helps so I could remove it.